### PR TITLE
switch code to java 11 and latest cbioportal

### PR DESCRIPTION
**Pull request by sheridancbio**
- codebase made consistent with java 11
  - Mockito update required
  - apache commons StringUtil dropped because of dependency clash
- switch to latest genome-nexus-annotation-pipeline version (brings in java 11 code in cbioportal dependency)
  - calls to annotator.getAnnotatedRecordsUsingPOST updated to include new required arguments
- whitespace and style cleanup

### CODE REVIEW

File: cvr/src/main/java/org/cbioportal/cmo/pipelines/cvr/mutation/CVRMutationDataReader.java [hunk 1]
```diff
@@ -188,7 +188,7 @@ private void annotateRecordsWithPOST(List<MutationRecord> records, boolean reann
         // annotate with GenomeNexusImpl annotator from genome nexus annotation pipeline
         // records will be partitioned inside annotator client
         // records which do not get a response back will automatically be defaulted to an AnnotatedRecord(record)
-        List<AnnotatedRecord> annotatedRecords = annotator.getAnnotatedRecordsUsingPOST(summaryStatistics, records, "mskcc", true, postIntervalSize, reannotate);
+        List<AnnotatedRecord> annotatedRecords = annotator.getAnnotatedRecordsUsingPOST(summaryStatistics, records, "mskcc", true, postIntervalSize, reannotate, "StripEntireSharedPrefix", Boolean.TRUE, Boolean.FALSE);
```
Comment (n1zea144) [inline cvr/src/main/java/org/cbioportal/cmo/pipelines/cvr/mutation/CVRMutationDataReader.java:191]: I guess the signature changed with the latest version of the anotator.

**Review by n1zea144 — APPROVED** [review]

File: cvr/src/main/java/org/cbioportal/cmo/pipelines/cvr/mutation/CVRMutationDataReader.java (hunk 1 above)
Comment (sheridancbio) [reply cvr/src/main/java/org/cbioportal/cmo/pipelines/cvr/mutation/CVRMutationDataReader.java:191]: That's correct. These three new arguments were set based on a reading of the genome-nexus-annotation-pipeline codebase.

File: cvr/src/main/java/org/cbioportal/cmo/pipelines/cvr/whitelist/ZeroVariantWhitelistTasklet.java [hunk 2]
```diff
@@ -70,7 +68,8 @@ public RepeatStatus execute(StepContribution sc, ChunkContext cc) throws Excepti
             Set<String> compiledWhitelistedSamplesWithZeroVariants = new HashSet<>(cvrSampleListUtil.getWhitelistedSamplesWithZeroVariants());
             compiledWhitelistedSamplesWithZeroVariants.addAll(cvrSampleListUtil.getNewUnreportedSamplesWithZeroVariants());
             File whitelistedSamplesFile = new File(stagingDirectory, CVRUtilities.ZERO_VARIANT_WHITELIST_FILE);
-            FileUtils.write(whitelistedSamplesFile, StringUtils.join(compiledWhitelistedSamplesWithZeroVariants, "\n") + "\n");
+            String contents = String.join("\n", compiledWhitelistedSamplesWithZeroVariants) + "\n";
```
[the remark this answers is not in the document]
Comment (sheridancbio) [reply cvr/src/main/java/org/cbioportal/cmo/pipelines/cvr/whitelist/ZeroVariantWhitelistTasklet.java:71]: I believe so. The order is switched between apache commons StringUtil.join and java's built in String.join.

apache commons takes the collection of strings first, then the delimiter:
https://commons.apache.org/proper/commons-lang/apidocs/org/apache/commons/lang3/StringUtils.html#join-java.lang.Iterable-java.lang.String-
the java builtin takes the delimiter first, then the collection of strings : https://docs.oracle.com/en%2Fjava%2Fjavase%2F11%2Fdocs%2Fapi%2F%2F/java.base/java/lang/String.html#join(java.lang.CharSequence,java.lang.Iterable)

**Comment by n1zea144** [comment]
Feel free to merge at your convenience

**Comment by sheridancbio** [comment]
merged. Thank you for review. I am rebasing #1060 now based on the new master branch.

**Pull request merged**